### PR TITLE
Micro-ROS client retry mechanism

### PR DESCRIPTION
**Pull request by BlueAndi**
#175
#176

### CODE REVIEW

File: lib/APPTurtle/src/MicroRosClient.cpp [hunk 1]
```diff
@@ -178,12 +124,39 @@ bool MicroRosClient::registerSubscriber(BaseSubscriber* subscriber)
...
+    case STATE_CONNECTED:
+        connectedState();
+        break;
+
+    case STATE_DISCONNECTED:
+        disconnectedState();
+        break;
+
+    default:
+        /* Should never happend. */
```
Comment (gabryelreyes) [inline lib/APPTurtle/src/MicroRosClient.cpp:155]: typo "happen"

**Review by gabryelreyes — APPROVED** [review]

**Review by gabryelreyes — APPROVED** [review]

File: lib/APPTurtle/src/CustomRosTransport.cpp [hunk 2]
```diff
@@ -77,10 +71,10 @@ bool CustomRosTransport::open(uxrCustomTransport* transport)
     }
     else
     {
-        const micro_ros_agent_locator* locator = reinterpret_cast<micro_ros_agent_locator*>(transport->args);
-        const int                      UDP_OK  = 1;
+        CustomRosTransport* tthis  = reinterpret_cast<CustomRosTransport*>(transport->args);
```
Comment (nhjschulz) [inline lib/APPTurtle/src/CustomRosTransport.cpp:74]: I never saw this  tthis name convention before and first thought it is a typo.
Can't we be more explicit and call it "transport" or "thisTransport" to be closer to naming CS.

**Review by nhjschulz — COMMENTED** [review]
optional comment on variable name

File: lib/APPTurtle/src/CustomRosTransport.cpp (hunk 2 above)
Comment (BlueAndi) [reply lib/APPTurtle/src/CustomRosTransport.cpp:74]: We can fix it with next feature integration, this PR is already merged.

**Pull request merged**